Dockerfile

```diff
@@ -0,0 +1,27 @@
+# Use a lightweight base image
+FROM node:18-alpine
+
+# Set the working directory
+WORKDIR /app
+
+# Copy the package.json and package-lock.json files
+COPY package*.json ./
+
+# Install dependencies and clean npm's cache
+RUN npm install --production && \
+    npm cache clean --force
+
+# Copy the application code
+COPY . .
+
+# Set the non-root user to run the application
+USER node
+
+# Expose the port that the application runs on
+EXPOSE 3000
+
+# Define a health check
+HEALTHCHECK --interval=30s --timeout=5s --start-period=5s --retries=3 CMD curl --fail http://localhost:3000/health || exit 1
+
+# Start the application
```

Comment:
there are probably should be steps for build ts, to not run ts in production.
It's better approach to use `tsc` as build step and then `node`.